docs/docs/hydro/live-collections/streams.md

```diff
@@ -3,10 +3,119 @@ sidebar_position: 2
+#     assert_eq!(stream.next().await, Some(w));
+# }
+# }));
+```
+
+Streams also can be sent over the network to form distributed programs. Under the hood, sending a stream sets up an RPC handler at the target location that will receive the stream elements. For example, we can send a stream of integers from one process to another with [bincode](https://docs.rs/bincode/latest/bincode/) serialization:
+
+```rust
+# use hydro_lang::*;
+# use dfir_rs::futures::StreamExt;
+# tokio_test::block_on(test_util::multi_location_test(|flow, p_out| {
+let p1 = flow.process::<()>();
+let numbers: Stream<_, Process<_>, _> = p1.source_iter(q!(vec![1, 2, 3]));
+let p2 = flow.process::<()>();
+let on_p2: Stream<_, Process<_>, _> = numbers.send_bincode(&p2);
+// 1, 2, 3
+# on_p2.send_bincode(&p_out)
+# }, |mut stream| async move {
+# for w in 1..=3 {
+#     assert_eq!(stream.next().await, Some(w));
+# }
+# }));
+```
+
+## Stream Ordering and Determinism
+When sending a stream over the network, there are certain situations in which the order of messages will not be deterministic for the receiver. For example, when sending streams from a cluster to a process, delays will cause messages from different cluster members to be interleaved in a non-deterministic order.
+
+To track this behavior, stream have an `Order` type parameter that indicates whether the elements in the stream will have a deterministic order ([`TotalOrder`](pathname:///rustdoc/hydro_lang/stream/struct.TotalOrder)) or not ([`NoOrder`](pathname:///rustdoc/hydro_lang/stream/struct.NoOrder)). When the type parameter is omitted, it defaults to `TotalOrder` for brevity.
+
+If we send a stream from a cluster to a process, the return type will be a stream with `NoOrder`:
+
+```rust,no_run
+# use hydro_lang::*;
+# let flow = FlowBuilder::new();
+let workers: Cluster<()> = flow.cluster::<()>();
+let numbers: Stream<_, Cluster<_>, _, TotalOrder> =
+    workers.source_iter(q!(vec![1, 2, 3]));
+let process: Process<()> = flow.process::<()>();
+let on_p2: Stream<_, Process<_>, _, NoOrder> =
+    numbers.send_bincode(&process);
+```
+
+The ordering of a stream determines which APIs are available on it. For example, `map` and `filter` are available on all streams, but `last` is only available on streams with `TotalOrder`. This ensures that even when the network introduces non-determinism, the program will not compile if it tries to use an API that requires a deterministic order.
+
+A particularly common API that faces this restriction is [`fold`](pathname:///rustdoc/hydro_lang/stream/struct.Stream#method.fold) (and [`reduce`](pathname:///rustdoc/hydro_lang/stream/struct.Stream#method.reduce)). These APIs require the stream to have a deterministic order, since the aggregation may depend on the order of elements. For example, the following code will not compile because `fold` is not available on `NoOrder` streams. The error is a bit misleading, but the key part is that `fold` is not available on `NoOrder` streams:
```

Comment:
"the aggregation" --> "the outcome of applying the Rust code in the closure"

Reply:
The confusing error code is unfortunate. Are there examples where we have better error codes? 

And is this error code an issue we should file?